envoy/http/filter.h

```diff
@@ -757,13 +757,14 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks,
    * host list of the routed cluster, the host should be selected first.
    * @param host The override host address.
    */
-  virtual void setUpstreamOverrideHost(absl::string_view host) PURE;
+  virtual void setUpstreamOverrideHost(absl::string_view host, bool strict) PURE;
```

Comment:
`virtual void setUpstreamOverrideHost(Upstream::LoadBalancerContext::OverrideHost host) PURE;`

Reply:
I actually left that intensionally in that form. The reason is that info about the desired upstream host is assembled from two distinct pieces: config option `strict` and IP address from decoded header. There must be a point in the code where those two pieces are assembled into `Upstream::LoadBalancerContext::OverrideHost`, which is a pair. I thought that it would be cleaner to assemble them in the `setUpstreamOverrideHost` method, but maybe it would be better to assemble them into a pair before calling that method. Let me experiment.

Reply:
Refactored.